core/domain/query_domain.py

```diff
@@ -0,0 +1,104 @@
+# coding: utf-8
+#
+# Copyright 2016 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Domain object for a parameters of a query."""
+
+from core.platform import models
+
+(job_models,) = models.Registry.import_models([models.NAMES.job])
+
+# pylint: disable=C0103
+class QueryParameters(object):
+    """Domain object for parameters of a query.
+
+    Each query has a combination of these parameters. This parameters are used
+    for checking whether given user qualifies a query or not.
+
+    Attributes:
+        has_user_logged_in_since_n_days: query parameter that specifies whether
```

Comment:
What's the type -- bool/int/None? I can't tell from the docstring.

When the type has been decided, suggest writing as follows:

       has_user_logged_in_since_n_days: int or None. If int, users satisfying the query should have logged in within the last n days. Otherwise, this option is not part of the query.

(see #2465 for more examples).